### ISSUE TITLE
This looks cool, you should implement this as an npm package!

### ISSUE THREAD

**Issue by nfischer** [Jeff-Russ/spelunker-node-shelljs#1]
This project seems promising, so you should go forward and implement this as an npm package! I'd like to see the source code moved out into `index.js` and this refactored as an `npm` module (I can help if you like) so that it's easier for others to try out.

Also, I noticed that you're using `eval()`, which could be a potential security risk, so I'd recommend putting a warning at the top of the README until you can refactor that out.

Does this actually rely on shelljs commands? It looks like it just uses `exec()`, which is a shame since I think you can take advantage of some of the other shelljs functions such as `cp()`, `cd()`, `pwd()`, etc.

**Comment by Jeff-Russ** [comment]
The entire thing is a huge security risk. lol. I don't think spelunker belongs in a web app, it's really for making installer scripts. It's part of my trying to transition from Bash to Node. I REALLY should put a warning in there now that you mention it, in case it's not apparent that having javascript force the shell to output raw javascript and then executing it as a string  doesn't tell the tale. Anyway I got the whole idea of building a large set of commands before sending from RoR's way of holding of on executing SQL. I'm glad you think it's promising! I'm really getting into shelljs as well!

**Comment by Jeff-Russ** [comment]
**my original response**  

I'm glad you think it's promising! I'm really getting into shelljs as well! Of course my little solution is a huge security risk. lol. I don't think spelunker belongs in a web app, it's really for making installer scripts. I REALLY should put a warning in there now that you mention it.

**the rest...**   

I put a warning in the README.md. I guess I totally ignored your suggestion/proposal about making this a node package, so here is some back-story.  

## tl;dr

This is just a tiny unsafe function used in a script I'm working on that initializes a new node (or other type of) project. It wasn't intended to be used in any actual project. That said, I wish shelljs had something like this and perhaps one day I'll dive into your source and see if I can think of a way to do it right.  

## Back Story

I'm making a script that replaces `node init` to be more like `rails new` but better. The idea is that you type:  

``` bash
$ appgen node ./newproj node-sass
$ #   generating new node project at '~/workspace/newproj' 
$ #   using '~/appgen-templates/node-sass' as boilerplate
```

and it clones a copy of `~/appgen-templates/node-sass`, generates a `package.json` based on information gathered from git/the OS and/or anything already in the destination directory and runs all `npm install` shell command needed to get you to the point where you can **just start coding**. `appgen` was originally a Bash script but I thought I should start rewriting it with Node.js lest it be some sort of admission of Node's shortcomming.    

## Shortcoming of Node.js as a Shell Language

Funny thing is _rewriting it with Node.js is totally revealing Node's shortcomings as a shell tool_ LOL. It's SUPER slow in Node. I have about 10 things I need to get from the shell and in the Node version take about as long as opening 10 tabs in terminal and firing off commands in each.  

I made this little function to run everything in one sub-shell and it totally worked so I made a little Gist for it. Then I thought: "I wish shelljs had some way of doing this" so I posted it to your repo to get your attention. heh.  

## Resolving this Repo Issue

Right now I'm making this `appgen` tool to make making Node projects easier and I don't want get _meta_-distracted into making a tool that makes a tool to make Node projects easier. There are also a few reason's my idea would not work in a Node package but, as I said, something like this but better/safer should be part of shelljs. One day, when I'm more seasoned, I'll dive into a fork of shelljs and see what might be possible.   

I'm gonna leave this issue open as I plan to re-visit it after finishing the larger project it's part of.

**Comment by Jeff-Russ** [comment]
I'm gonna leave this issue open as I plan to re-visit it after finishing the larger project it's part of.

**Comment by Jeff-Russ** [comment]
Looks like I just invented a term:  

## meta-distracted

**adjective** 
1. having the attention diverted from creating a technique or system for making a primary task easier to perform by creating another technique to make the creation if the first technique or system easier. **example**
"I started to make a Javascript library that makes making forms easier but it was taking too long so I got meta-distracted and starting making a another library to help in making the first. Then I got meta-meta-distracted by the fact that I came up with the term "meta-distracted" and made a post about it online."

**Comment by Jeff-Russ** [comment]
Compromise:  

I made it `require`-abe with:  

``` javascript
require('./path/to/spelunker.js').globalize();
```

add added instructions to "install" with 

``` bash
$ curl -O https://raw.githubusercontent.com/Jeff-Russ/spelunker-node-shelljs/master/spelunker.js
$ curl -O https://raw.githubusercontent.com/Jeff-Russ/spelunker-node-shelljs/master/spelunker.man
$ ./spelunker.man  # displays manual
```

So that'll be the "node package" for now.
